lib/puppet/util/windows/eventlog.rb

```diff
@@ -0,0 +1,114 @@
+require 'puppet/util/windows'
```

Comment:
The simplest way of making this consumable inside the service daemon is to start by removing the above `require 'puppet/util/windows'` and see what breaks.

Things I know of off the top of my head:

* You'll need an in-file version of `Puppet::Util::Windows::String.wide_string` - which is pretty easy
* `FFI::Pointer::NULL_HANDLE` comes from `api_types.rb`
* `Puppet::Util::Windows::Error` will have to be replaced OR we might be able to make that one usable within the service daemon as well as it's quite small and self-contained
* `FFI::WIN32_FALSE` comes from `api_types.rb`
* `FFI::MemoryPointer.from_string_to_wide_string` comes from `api_types.rb`
* `attach_function_private` comes from `api_types.rb`
* a bunch of the aliases in the API prototypes come from `api_types.rb`


It's also possible that we can make both `api_types.rb`and `windows/string.rb` consumable without the rest of Puppet (`string.rb` currently requires `puppet/util/windows` but I'm not sure why)

Food for thought anyhow...